src/components/AddressHelp.js

```diff
@@ -7,24 +7,60 @@ import { operatorTypeEnum } from "@slice-and-dice/register-a-food-business-valid
 
 const AddressHelp = (props) => {
   return props.role === operatorTypeEnum.PARTNERSHIP.key ? (
+    props.id ? (
+      <React.Fragment>
```

Comment:
Instead of duplicating the whole operator section and whole partnership section, you could have the props.id check just on the Heading where it matters.  It might make it easier to read and less duplication.  E.g.:
```
return props.role === operatorTypeEnum.PARTNERSHIP.key ? (
      <React.Fragment>
        <Heading as="h1" size="LARGE">
props.id ? (
          <label class="govuk-label govuk-label--l" for={props.id}>
            {props.t("What is the partnership contact's address?")}
          </label>
) : (
{props.t("What is the partnership contact's address?")}
)
       </Heading>
        <ContentItem.B_30_15>
          <HintText>
            {props.t(
              "Partnership address is the contact address for the partner who is the main point of contact."
            )}
          </HintText>
        </ContentItem.B_30_15>
        <PartnershipDescription />
      </React.Fragment>
    ) : ...
```